web_scraping/steam_scraper/scrape_steam.py

```diff
@@ -67,20 +68,85 @@ def parse_price(game_listing: bs4.Tag) -> str:
     return format_price(price)
 
 
+def get_steam_app_url(app_id: str) -> str:
+    """Formats the base steam url for a particular app."""
+    return f"{STEAM_APP_URL}{app_id}/"
+
+
+def scrape_game_description(app_soup: bs4.BeautifulSoup) -> str:
+    """Scrapes a game's Steam app page for description information."""
+    description_tag = app_soup.find('div', class_='game_description_snippet')
+    if description_tag:
+        return description_tag.text.strip()
+    return "No description available."
+
+
+def scrape_game_tags(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for tags."""
+    tag_elements = app_soup.find_all('a', class_='app_tag')
+    tags = [tag.text.strip() for tag in tag_elements]
+    return tags if tags else ["No tags available."]
+
+
+def scrape_game_genres(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for genres."""
+    genres_section = app_soup.find('div', id='genresAndManufacturer')
+    if genres_section:
+        genre_elements = genres_section.find_all('a', href=True)
+        genres = [genre.text.strip()
+                  for genre in genre_elements if "genre" in genre['href']]
+    return genres if genres else []
+
+
+def scrape_game_operating_systems(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for supported operating systems."""
+    platform_mapping = {
+        'win': 'Windows',
+        'mac': 'Mac',
+        'linux': 'Linux'
```

Comment:
Could go even further and only look at the first character lol

Reply:
Good idea. I believe dictionaries are constant lookup time regardless, though. 💪